hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -278,18 +275,17 @@ private ContainerInfo allocateContainer(final Pipeline pipeline,
         .build();
     containerStateManager.addContainer(containerInfo);
     scmContainerManagerMetrics.incNumSuccessfulCreateContainers();
-    return containerStateManager.getContainer(containerID.getProtobuf());
+    return containerStateManager.getContainer(containerID);
   }
 
   @Override
-  public void updateContainerState(final ContainerID id,
+  public void updateContainerState(final ContainerID cid,
                                    final LifeCycleEvent event)
       throws IOException, InvalidStateTransitionException {
-    final HddsProtos.ContainerID cid = id.getProtobuf();
     lock.lock();
     try {
       if (containerExist(cid)) {
-        containerStateManager.updateContainerState(cid, event);
+        containerStateManager.updateContainerState(cid.getProtobuf(), event);
```

Comment:
Is it worth moving the `getProtobuf()` call inside the locked section, to avoid calls for nonexistent containers?

Reply:
For this one, the `getProtobuf()` call is already inside the locked section, or do I not understand the issue?

Reply:
> For this one, the `getProtobuf()` call is already inside the locked section,

Yes, that's what I'd like toavoid if possible, since container ID need not be guarded by the lock during conversion.  I was wondering about the pros/cons:

 * outside the lock: called eagerly (even if container does not exist), but does not hold lock
 * inside lock: may avoid some unnecessary calls, but increases the time lock is held

Reply:
I see your point. I'd guess the number of "not found" containers would be very low, so it probably makes sense to lift the protobuf overhead out of the lock.